docs: Sustainability effort (Incentive system) (#12359)
argoproj/argo-workflows: docs/CONTRIBUTING.md

## Contributor Meetings (twice monthly)

> An opportunity for committers and maintainers of Workflows and Events to discuss their current work and
> talk about what’s next. Feel free to join us! For Contributor Meeting information, minutes and recordings
> please [see here](https://bit.ly/argo-data-weekly).

Reviewer: nice catch! I noticed this was no longer correct at some point too and forgot to fix it 😅 

while we're at it, could we get this paragraph to match the docs style guide too?

```suggestion
An opportunity for contributors and maintainers of Workflows and Events to discuss their current work and talk about what’s next. Feel free to join us!
See the [Contributor Meeting doc](https://bit.ly/argo-data-weekly) for minutes, recordings, and more information.
```

## Slack

You can join the following channels on [CNCF Slack](https://argoproj.github.io/community/join-slack):
* `#argo-workflows`: discussions focused mainly on use of Argo Workflows
* `#argo-wf-contributors`: discussions focused mainly on development of Argo Workflows

## How To Contribute

We're always looking for contributors.

> ### Roles

Reviewer: So the "Roles" section and the "Sustainability Effort" are actually not specific to "How To Contribute" (the "how" in particular), so I wouldn't put them as subsections of it. I'm fine with them either being above or below the entire "How To Contribute" section

## Contributor Meetings (twice monthly)

An opportunity for contributors and maintainers of Workflows and Events to discuss their current work and talk about what’s next. Feel free to join us!
See the [Contributor Meeting doc](https://bit.ly/argo-data-weekly) for minutes, recordings, and more information.

## Slack

You can join the following channels on [CNCF Slack](https://argoproj.github.io/community/join-slack):

> * `#argo-workflows`: discussions focused mainly on use of Argo Workflows
> * `#argo-wf-contributors`: discussions focused mainly on development of Argo Workflows

Reviewer: ```suggestion
* [`#argo-workflows`](https://cloud-native.slack.com/archives/C01QW9QSSSK): discussions focused mainly on use of Argo Workflows
* [`#argo-wf-contributors`](https://cloud-native.slack.com/archives/C0510EUH90V): discussions focused mainly on development of Argo Workflows
```

add direct links to the channel same as other sections in this doc

## Roles

The Argo project currently has 4 designated [roles](https://github.com/argoproj/argoproj/blob/main/community/membership.md):

* Member
* Reviewer
* Approver
* Lead

The Reviewer and Approver roles can optionally be scoped to an area of the code base (for example, UI or docs).

Current roles for Reviewers and above can be found in [OWNERS](../OWNERS).

> If you are interested in formally joining the Argo project, [create a Membership request](https://github.com/argoproj/argoproj/issues/new?template=membership.md&title=REQUEST%3A%20New%20membership%20for%20%3Cyour-GH-handle%3E) in the [argoproj](https://github.com/argoproj/argoproj) repository as described in the [Membership](https://github.com/argoproj/argoproj/blob/main/community/membership.md) guide.

Reviewer: ```suggestion
If you are interested in formally joining the Argo project, [create a Membership request](https://github.com/argoproj/argoproj/issues/new?template=membership.md&title=REQUEST%3A%20New%20membership%20for%20%3Cyour-GH-handle%3E) in the [`argoproj` repository](https://github.com/argoproj/argoproj) as described in the [Membership](https://github.com/argoproj/argoproj/blob/main/community/membership.md) guide.
```

we could put `argoproj` in backticks, and then we could remove it from the `.spelling`. nbd either way though, up to you

Author: I'll just leave as is

  a [good first issue](https://github.com/argoproj/argo-workflows/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
  , or anything not assigned.
* You can work on an issue without being assigned.

> * Join the `#argo-wf-contributors` channel (and optionally the `#argo-contributors` channel for all Argo projects) on [our Slack](https://argoproj.github.io/community/join-slack)

Reviewer: ```suggestion
* Join the `#argo-wf-contributors` channel (and optionally the `#argo-contributors` channel for all Argo projects) on the [CNCF Slack](https://argoproj.github.io/community/join-slack)
```

"our Slack" was probably outdated language from when Argo had its own Slack

No, we should not add that dependency.

#### Test Policy

Changes without either unit or e2e tests are unlikely to be accepted.
See [the pull request template](https://github.com/argoproj/argo-workflows/blob/main/.github/pull_request_template.md).

### Other Contributions

* [Reviewing PRs](#reviewing-prs)
* Responding to questions in the [Slack](#slack) channels
* Responding to questions in [Github Discussions](https://github.com/argoproj/argo-workflows/discussions)
* [Triaging new bugs](#triaging-bugs)

#### Reviewing PRs

> Anybody can review a PR; however, those who are in the roles of Reviewer and above are expected to assign themselves to PRs.
> Since Reviewers cannot merge PRs, then once they have approved a PR, they should request a review from one or more Approvers/Leads.

Reviewer: ```suggestion
Anybody can review a PR.
If you are in a [designated role](#roles), add yourself as an "Assignee" to a PR if you plan to lead the review.
If you are a Reviewer or below, then once you have approved a PR, request a review from one or more Approvers and above.
```

- address the reader as "you", per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#address-the-reader-as-you)
  - also matches the existing verbiage of this doc
  - also made the tone gentler as the Contributing Guide is giving directions, which is a different tone from "you are expected to" (which is more accurate for the incentive system, which has specific requirements)
  
- simplify the wording a bit (e.g. removed "you should"), per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
  
- linked to "Roles" section as specific roles are referenced here
  
- changed it to Reviewer and below; I certainly led review of various PRs as a Member and we have other contributors and Members who do so occasionally as well. if they're doing it already and can assign themselves, then why not do so I figure

No, we should not add that dependency.

#### Test Policy

Changes without either unit or e2e tests are unlikely to be accepted.
See [the pull request template](https://github.com/argoproj/argo-workflows/blob/main/.github/pull_request_template.md).

### Other Contributions

* [Reviewing PRs](#reviewing-prs)
* Responding to questions in the [Slack](#slack) channels
* Responding to questions in [Github Discussions](https://github.com/argoproj/argo-workflows/discussions)
* [Triaging new bugs](#triaging-bugs)

#### Reviewing PRs

Anybody can review a PR.
If you are in a [designated role](#roles), add yourself as an "Assignee" to a PR if you plan to lead the review.
If you are a Reviewer or below, then once you have approved a PR, request a review from one or more Approvers and above.

#### Triaging Bugs

New bugs need to be triaged to identify the highest priority ones.

Apply the labels `P0`, `P1`, `P2`, and `P3`, where `P0` is highest priority and needs immediate attention, followed by `P1`, `P2`, and then `P3`.

> If there's a new `P0` bug, notify the [#argo-wf-contributors](https://cloud-native.slack.com/archives/C0510EUH90V) slack channel.

Reviewer: ```suggestion
If there's a new `P0` bug, notify the [`#argo-wf-contributors`](https://cloud-native.slack.com/archives/C0510EUH90V) Slack channel.
```

For consistency with the "Slack" section above. This will also make the `.spelling` addition no longer needed
